test/lib/next-modes/base.ts

```diff
@@ -88,16 +88,22 @@ export class NextInstance {
     ) {
       await fs.copy(process.env.NEXT_TEST_STARTER, this.testDir)
     } else if (!skipIsolatedNext) {
+      const reactVersion = process.env.NEXT_TEST_REACT_VERSION || 'latest'
+      const finalDependencies = {
+        react: reactVersion,
+        'react-dom': reactVersion,
+        ...this.dependencies,
+        ...((this.packageJson.dependencies as object | undefined) || {}),
+      }
       this.testDir = await createNextInstall(
-        {
-          react: 'latest',
-          'react-dom': 'latest',
-          ...this.dependencies,
-          ...((this.packageJson.dependencies as object | undefined) || {}),
-        },
+        finalDependencies,
         this.installCommand,
         this.packageJson
       )
+      console.log('Using react version', {
+        env: reactVersion,
+        actual: finalDependencies.react,
```

Comment:
These will always be equal right?

You might want to read package.json here to get "actual"

Reply:
A specific test can override the react version by specifying it in the `dependencies` for `createNext` manually.